specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json

```diff
@@ -0,0 +1,554 @@
+      "description": "An Application Insights private workbook definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MyWorkbookResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The kind of workbook. Choices are user and shared.",
+          "enum": [
+            "user",
+            "shared"
+          ],
+          "x-ms-enum": {
+            "name": "SharedTypeKind",
+            "modelAsString": true
+          }
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook for an Azure resource.",
+          "$ref": "#/definitions/MyWorkbookProperties"
+        }
+      }
+    },
+    "MyWorkbookProperties": {
+      "description": "Properties that contain a private workbook.",
+      "required": [
+        "displayName",
+        "kind",
+        "category",
+        "serializedData",
+        "userId"
+      ],
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The user-defined name of the private workbook."
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular private workbook. Configuration data is a string containing valid JSON"
+        },
+        "version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked private workbook."
+        },
+        "kind": {
+          "description": "Enum indicating if this workbook definition is owned by a specific user or is shared between all users with access to the Application Insights component.",
+          "x-ms-client-name": "SharedTypeKind",
+          "default": "shared",
+          "type": "string",
+          "enum": [
+            "shared",
+            "user"
+          ],
+          "x-ms-enum": {
+            "name": "SharedTypeKind",
+            "modelAsString": true
+          }
+        },
+        "timeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this private workbook definition.",
+          "readOnly": true
+        },
+        "category": {
+          "type": "string",
+          "description": "Workbook category, as defined by the user at creation time."
+        },
+        "tags": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this private workbook definition"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Unique user id of the specific user that owns this private workbook."
+        },
+        "sourceId": {
+          "type": "string",
+          "description": "Optional resourceId for a source resource."
```

Comment:
Which kinds of resource are supported?

Reply:
any/all, and it doesn't even need to be a real resource id.  it also technically supports things that aren't real resources, like "Azure monitor" and other areas that are cross cutting like azure backup, azure active directory, azure sentinel, etc.